lib/seahorn/BvOpSem2ExtraWideMemMgr.cc

```diff
@@ -62,9 +62,11 @@ Expr ExtraWideMemManager<T>::isMetadataSet(MetadataKind kind,
 template <class T>
 Expr ExtraWideMemManager<T>::ptrEq(ExtraWideMemManager::PtrTy p1,
                                    ExtraWideMemManager::PtrTy p2) const {
-  return mk<AND>(m_main.ptrEq(p1.getBase(), p2.getBase()),
-                 m_offset.ptrEq(p1.getOffset(), p2.getOffset()),
-                 m_size.ptrEq(p1.getSize(), p2.getSize()));
+  // NOTE: we consider two pointers to be same if their address (base + ofset)
+  //       is the same. Size is ignored. This is done to have parity with memset
+  //       like operations that zero out main memory but do not touch shadow
+  //       memory.
+  return m_main.ptrEq(getAddressable(p1), getAddressable(p2));
```

Comment:
this is cleaner but less efficient than the previous version. Previous version would create `p1.base == p2.base && p1.offset == p2.offset`, the new version is `p1.base + p1.offset == p2.base + p2.offset`. 

pointers with different bases might be equal. This, at the very least, will create more work to the smt solver.

Reply:
Fixed.